products/surveys/backend/max_tools.py

```diff
@@ -167,3 +151,133 @@ def _prepare_survey_data(self, survey_schema: SurveyCreationSchema, team: Team)
         survey_data["appearance"] = appearance
 
         return survey_data
+
+
+class FeatureFlagToolkit(TaxonomyAgentToolkit):
+    """Toolkit for feature flag lookup operations."""
+
+    def __init__(self, team: Team):
+        super().__init__(team)
+        self._last_lookup_result: FeatureFlagLookupResult | None = None
+
+    def get_tools(self) -> list:
+        """Get all tools (default + custom). Override in subclasses to add custom tools."""
+        return self._get_custom_tools()
+
+    def _get_custom_tools(self) -> list:
+        """Get custom tools for feature flag lookup."""
+
+        class lookup_feature_flag(BaseModel):
+            """
+            Use this tool to lookup a feature flag by its key/name to get detailed information including ID and variants.
+            Returns a message with the flag ID and the variants if the flag is found and the variants are available.
+            """
+
+            flag_key: str = Field(description="The key/name of the feature flag to look up")
+
+        class final_answer(base_final_answer[SurveyCreationSchema]):
+            __doc__ = base_final_answer.__doc__
+
+        return [lookup_feature_flag, final_answer]
+
+    def handle_tools(self, tool_name: str, tool_input) -> tuple[str, str]:
+        """Handle custom tool execution."""
+        if tool_name == "lookup_feature_flag":
+            result = self._lookup_feature_flag(tool_input.arguments.flag_key)
+            return tool_name, result
+        return super().handle_tools(tool_name, tool_input)
+
+    def _lookup_feature_flag(self, flag_key: str) -> str:
+        """Look up feature flag information by key."""
+        try:
+            # Look up the feature flag by key for the current team
+            feature_flag = FeatureFlag.objects.select_related("team").get(key=flag_key, team_id=self._team.id)
```

Comment:
```suggestion
            feature_flag = FeatureFlag.objects.get(key=flag_key, team_id=self._team.id)
```

`select_related("team")` is unnecessary here since we're filtering by `team_id` and don't use the team object